### PR TITLE
Procedure parameter

### PR DESCRIPTION
**Pull request by sticmac**
Depending on #64.
Each procedure takes one parameter, corresponding to the address to go to for executing its content.

### CODE REVIEW

File: src/main/java/fr/unice/polytech/si3/miaou/brainfuck/instructions/Procedure.java [hunk 1]
```diff
@@ -32,7 +54,21 @@ public Procedure(String name, int position) {
...
 		machine.setInstrPointer(position-1); //-1, because the instruction pointer is incremented right after the execution of the Procedure instruction.
 	}
+
+	/**
+	 * Clones the current procedure instruction.
+	 *
+	 * @return the cloned procedure instruction.
+	 */
+	@Override
+	public Procedure clone() {
```
Comment (piernov) [inline src/main/java/fr/unice/polytech/si3/miaou/brainfuck/instructions/Procedure.java]: It is generally discouraged to override clone() method. Use a copy constructor instead. Also it doesn't seem to be used anywhere.

File: src/main/java/fr/unice/polytech/si3/miaou/brainfuck/parser/FunctionsParser.java [hunk 2]
```diff
@@ -70,6 +77,9 @@ public FunctionsParser(InstructionSet iset) {
 				throw new SyntaxFunctionException("Trying to declare a new function before the end of another one.");
 			} else {
 				name = line.split(" ")[1];
+				if (split.length > 2) {
```
Comment (piernov) [inline src/main/java/fr/unice/polytech/si3/miaou/brainfuck/parser/FunctionsParser.java]: Is there anything happening here?

File: src/main/java/fr/unice/polytech/si3/miaou/brainfuck/parser/InstructionParser.java [hunk 3]
```diff
@@ -10,6 +11,7 @@
 import fr.unice.polytech.si3.miaou.brainfuck.InstructionSet;
 import fr.unice.polytech.si3.miaou.brainfuck.JumpTable;
 import fr.unice.polytech.si3.miaou.brainfuck.exceptions.InvalidInstructionException;
+import fr.unice.polytech.si3.miaou.brainfuck.instructions.Procedure;
```
Comment (piernov) [inline src/main/java/fr/unice/polytech/si3/miaou/brainfuck/parser/InstructionParser.java]: Some useless imports apparently.

File: src/main/java/fr/unice/polytech/si3/miaou/brainfuck/parser/InstructionTextParser.java [hunk 4]
```diff
@@ -36,6 +36,9 @@ public void accept(String line) {
 			addAndBindInstructions(instr);
 		} else if (iset.getProc(split[0]) != null) {
 			Procedure proc = iset.getProc(split[0]);
+			if (split.length > 1) {
```
Comment (piernov) [inline src/main/java/fr/unice/polytech/si3/miaou/brainfuck/parser/InstructionTextParser.java]: The whole `else if (iset.getProc(split[0]) != null)` should probably be done in a method of (let's say) FunctionParser.

**Review by piernov — COMMENTED** [review]
Also do not forget to add documentation about function declaration&usage, and possible incompatibilities (with image syntax for example).

**Review by NassimBounouas — APPROVED** [review]

**Pull request merged**